region/maridia/inner-pink/The Precious Room.json

```diff
@@ -785,44 +757,102 @@
+    {
+      "link": [2, 4],
+      "name": "Stutter Water Shinecharge",
+      "entranceCondition": {
+        "comeInStutterShinecharging": {
+          "minTiles": 4
         }
       },
       "requires": [
         "canStutterWaterShineCharge",
-        "canShinechargeMovementComplex",
-        "h_canShineChargeMaxRunway",
-        {"or": [
-          {"shinespark": {"frames": 12}},
-          {"and": [
-            "canShinechargeMovementTricky",
-            {"shinespark": {"frames": 3}}
-          ]}
-        ]}
+        {"shineChargeFrames": 0}
       ],
-      "exitCondition": {
-        "leaveWithSpark": {}
+      "endsWithShineCharge": true,
+      "note": "Use between 4 and 5 tiles of runway, to make the stutter as lenient as possible.",
+      "detailNote": [
+        "With this length of runway, the following timings for the stutter are possible:",
+        "1. Release forward for 3, 4, or 5 frames, repress forward on the last possible frame;",
+        "2. Release forward for 2 or 3 frames, repress forward on the 2nd-to-last possible frame;",
+        "3. Release forward for 1 pr 2 frames, repress forward on the 3rd-to-last possible frame."
+      ]
+    },
+    {
+      "id": 15,
+      "link": [2, 4],
+      "name": "Precise Stutter Water Shinecharge",
+      "entranceCondition": {
+        "comeInStutterShinecharging": {
+          "minTiles": 2.4375
+        }
       },
-      "unlocksDoors": [
-        {"types": ["missiles", "super"], "requires": []},
-        {"types": ["powerbomb"], "requires": ["never"]}
+      "requires": [
+        "canPreciseStutterWaterShineCharge",
+        {"shineChargeFrames": 0}
       ],
-      "note": "Enter the room with a careful amount of momentum where Samus must spend extra time slowing down during the stutter, but must not have much run speed."
+      "endsWithShineCharge": true,
+      "note": [
+        "With only a runway length of 3 tiles (closed end) in the other room, this requires a precise setup:",
+        "release forward for 3 or 4 frames, then repress forward on the last possible frame before the transition;",
+        "alternatively, release forward for 2 frames and repress forward on the second-to-last possible frame before the transition."
+      ]
     },
     {
-      "id": 31,
-      "link": [2, 2],
-      "name": "Crystal Flash",
+      "link": [2, 4],
+      "name": "Tricky Stutter Water Shinecharge",
+      "entranceCondition": {
+        "comeInStutterShinecharging": {
+          "minTiles": 2
+        }
+      },
       "requires": [
-        "h_canCrystalFlash"
+        "canTrickyStutterWaterShineCharge",
+        {"shineChargeFrames": 0}
       ],
-      "flashSuitChecked": true
+      "endsWithShineCharge": true,
+      "note": [
+        "With only a runway of 2 tiles (open end) in the other room, this requires a double-frame-perfect setup:",
+        "release forward for exactly 3 frames, and repress forward on the last possible frame before the transition."
```

Comment:
I couldnt get this one to work even with frame advance. Not sure how I'm counting it differently

Reply:
![image_2025-03-19_141308449](https://github.com/user-attachments/assets/dca0fc7d-f5dd-4670-8e14-2e10207cc43b)

This is pretty much how I've been doing the trick and I feel better at it than double frame perfect would suggest.  The 3 tile version is less normalized and doesn't feel different enough for me.  But it could just be a me thing.